app_dart/lib/src/service/build_status_provider/commit_tasks_status.dart

```diff
@@ -50,12 +50,26 @@ final class CommitTasksStatus {
           didAtLeastOneFailureOccur: tasks.any(
             (t) => Task.taskFailStatusSet.contains(t.status),
           ),
-          lastCompletedAttemptWasFailure:
-              tasks.length > 1 &&
-              Task.taskFailStatusSet.contains(tasks[1].status),
+          lastCompletedAttemptWasFailure: _lastCompletedAttemptWasFailure(
```

Comment:
not that we should do this know, but this could be an enum for "lastCompletedStatus {success, failure, unknown}